lib/common/memory/src/madvise.rs

```diff
@@ -112,13 +112,15 @@ impl Madviseable for memmap2::Mmap {
         Ok(())
     }
 
-    fn populate(&self) -> io::Result<()> {
+    fn populate(&self) {
         #[cfg(target_os = "linux")]
         if *POPULATE_READ_IS_SUPPORTED {
-            return self.advise(memmap2::Advice::PopulateRead);
+            match self.advise(memmap2::Advice::PopulateRead) {
+                Ok(()) => return,
+                Err(err) => log::warn!("Failed to populate with MADV_POPULATE_READ: {err}"),
```

Comment:
Maybe it's worth to describe it's falling back:

```suggestion
                Err(err) => log::warn!("Failed to populate with MADV_POPULATE_READ, falling back to naive approach: {err}"),
```

Same goes for the other instance.

What do you think?

Reply:
Updated.

Reply:
Thanks!